debug-tools/gce-serial-output-continuously-append.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash -e
+
+INSTANCE="$1"
+
+if [ -z "$INSTANCE" ]; then
+    echo "first argument must be the GCE instance"
+    exit 1
+fi
+
+# wait for the instance to exist
+until gcloud compute --project=snapd-spread instances describe "$INSTANCE" --zone=us-east1-b >/dev/null 2>&1; do 
+    echo "waiting for instance to exist"
+    sleep 1
+done
+
+backgroundscript=/tmp/background-$RANDOM.sh
+cat >> "$backgroundscript" << 'EOF'
+INSTANCE="$1"
+next=0
+truncate -s0 console-output.txt
+while true; do
+    next=$(
+        gcloud compute \
+            --project=snapd-spread \
+            instances get-serial-port-output "$INSTANCE" \
+            --start="$next" \
+            --zone=us-east1-b 3>&1 1>"console-output-bits.txt" 2>&3- | grep -Po -- '--start=\K[0-9]+')
```

Comment:
We should have a comment explaining this line:
1. Explain what the file descriptor redirections do
2. Show an example of the stderr of the command, to show why we are using that `grep` expression

Reply:
Added!